mne/viz/evoked.py

```diff
@@ -1448,6 +1449,10 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     elif not isinstance(evokeds, dict):
         evokeds = dict((str(ii + 1), evoked)
                        for ii, evoked in enumerate(evokeds))
+    for cond in evokeds.keys():
+        if not isinstance(cond, str):
```

Comment:
`string_types` you know this @jona-sassenhagen, must have been too long since your last PR...